main.go

```diff
@@ -453,25 +505,220 @@ func (c *controller) waitForCacheSync(stop <-chan struct{}) error {
+		initContainerTemplate.EnvFrom = []corev1.EnvFromSource{
+			{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: sts.Labels[envVarSecretLabelKey],
+					},
+				},
+			},
+		}
+	}
+
+	helper := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cleanup-" + name,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy:  corev1.RestartPolicyNever,
+			InitContainers: make([]corev1.Container, len(sts.Spec.VolumeClaimTemplates)),
+			Containers: []corev1.Container{
+				{
+					Name:            "cleanup",
+					Image:           c.options.cleanupImage,
+					Command:         []string{"rm", "-rf"},
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					VolumeMounts:    make([]corev1.VolumeMount, len(sts.Spec.VolumeClaimTemplates)),
+				},
+			},
+			Volumes: make([]corev1.Volume, len(sts.Spec.VolumeClaimTemplates)),
+		},
+	}
+	var v corev1.Volume
+	var vname, mountPath string
+	for i, t := range sts.Spec.VolumeClaimTemplates {
+		// Create a new Volu,e for this template.
+		vname = fmt.Sprintf("%s-%s", t.Name, name)
+		v = corev1.Volume{
+			Name: vname,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: vname,
+				},
+			},
+		}
+		helper.Spec.Volumes[i] = v
+		// Create an init container to replicate this Volume.
+		helper.Spec.InitContainers[i] = *initContainerTemplate.DeepCopy()
+		helper.Spec.InitContainers[i].VolumeMounts = []corev1.VolumeMount{{
+			Name:      v.Name,
+			MountPath: fromMountPath,
+		}}
+		// Add this Volume to the container that removes files.
+		mountPath = filepath.Join("/pvc", vname)
+		helper.Spec.Containers[0].VolumeMounts[i] = corev1.VolumeMount{
+			Name:      v.Name,
+			MountPath: mountPath,
+		}
+		helper.Spec.Containers[0].Command = append(helper.Spec.Containers[0].Command, filepath.Join(mountPath, "*"))
+	}
+
+	return helper
+}
+
+func (c *controller) cleanUp(pod *corev1.Pod) error {
+	c.cleanupAttempts.Inc()
+	sts, err := c.resolvePodOwnerRef(pod.Namespace, pod.GetOwnerReferences())
+	if err != nil {
+		c.cleanupErrors.WithLabelValues(fetchLabel).Inc()
+		return errors.Wrap(err, "could not get StatefulSet")
+	}
+	// This probably means that the Pod did not belong to a StatefulSet with
+	// our label selector, i.e. not a StatefulSet we are watching.
+	if sts == nil {
+		return nil
+	}
+	// Nothing to clean up.
+	if len(sts.Spec.VolumeClaimTemplates) == 0 {
+		return nil
+	}
+
+	_, secretOK := sts.Labels[objstoreSecretLabelKey]
+	_, keyOK := sts.Labels[objstoreSecretKeyLabelKey]
+	if !secretOK || !keyOK {
+		c.cleanupErrors.WithLabelValues(decodeLabel).Inc()
+		return fmt.Errorf("StatefulSet %s/%s is missing either the %s or %s label", sts.Namespace, sts.Name, objstoreSecretLabelKey, objstoreSecretKeyLabelKey)
+	}
+	helper := c.generateHelper(pod.Name, sts)
+
+	if _, err := c.klient.CoreV1().Pods(pod.Namespace).Create(helper); err != nil && !kerrors.IsAlreadyExists(err) {
```

Comment:
seems like this should be a `Job` no?

Reply:
Yeah I feel like a `Job` is the appropriate thing here.

Reply:
done :)